go-apps/meep-loc-serv/server/loc-serv.go

```diff
@@ -384,215 +388,235 @@ func checkNotificationDistancePeriodicTrigger() {
-						//need pair to be symmetrical so that each is used as reference point and monitored address
-						pair = Pair{addr1: distanceCheck.Subscription.MonitoredAddress[j], addr2: distanceCheck.Subscription.MonitoredAddress[i]}
-						addressPairs = append(addressPairs, pair)
+				//loop through every reference address
+				returnAddr := make(map[string]*gisClient.Distance)
+				skipThisSubscription := false
+
+				//if reference address is specified, reference addresses are checked agains each monitored address
+				//if reference address is nil, each pair of the monitored address should be checked
+				//creating address pairs to check
+				//e.g. refAddr = A, B ; monitoredAddr = C, D, E ; resultingPairs {A,C - A,D - A,E - B,C - B,D - B-E}
+				//e.g. monitoredAddr = A, B, C ; resultingPairs {A,B - B,A - A,C - C,A - B,C - C,B}
+
+				var addressPairs []Pair
+				if distanceCheck.Subscription.ReferenceAddress != nil {
+					for _, refAddr := range distanceCheck.Subscription.ReferenceAddress {
+						//loop through every monitored address
+						for _, monitoredAddr := range distanceCheck.Subscription.MonitoredAddress {
+							pair := Pair{addr1: refAddr, addr2: monitoredAddr}
+							addressPairs = append(addressPairs, pair)
+						}
+					}
+				} else {
+					nbIndex := len(distanceCheck.Subscription.MonitoredAddress)
+					for i := 0; i < nbIndex-1; i++ {
+						for j := i + 1; j < nbIndex; j++ {
+							pair := Pair{addr1: distanceCheck.Subscription.MonitoredAddress[i], addr2: distanceCheck.Subscription.MonitoredAddress[j]}
+							addressPairs = append(addressPairs, pair)
+							//need pair to be symmetrical so that each is used as reference point and monitored address
+							pair = Pair{addr1: distanceCheck.Subscription.MonitoredAddress[j], addr2: distanceCheck.Subscription.MonitoredAddress[i]}
+							addressPairs = append(addressPairs, pair)
+						}
 					}
 				}
-			}
-
-			for _, pair := range addressPairs {
-				refAddr := pair.addr1
-				monitoredAddr := pair.addr2
 
-				var distParam gisClient.TargetPoint
-				distParam.AssetName = monitoredAddr
-
-				distResp, _, err := gisAppClient.GeospatialDataApi.GetDistanceGeoDataByName(context.TODO(), refAddr, distParam)
-				if err != nil {
-					log.Error("Failed to communicate with gis engine: ", err)
-					return
-				}
+				for _, pair := range addressPairs {
+					refAddr := pair.addr1
+					monitoredAddr := pair.addr2
+
+					var distParam gisClient.TargetPoint
+					distParam.AssetName = monitoredAddr
+
+					distResp, _, err := gisAppClient.GeospatialDataApi.GetDistanceGeoDataByName(context.TODO(), refAddr, distParam)
+					if err != nil {
+						//getting distance of an element that is not in the DB (not in scenario, not connected) returns error code 400 (bad parameters) in the API. Using that error code to track that request made it to GIS but no good result, so ignore that address (monitored or ref)
+						errCodeStr := strings.Split(err.Error(), " ")
```

Comment:
You should retrieve the status code from the http response rather than parsing the error string.

Reply:
fixed